### ISSUE TITLE
pre-training

### ISSUE THREAD

**Issue by code666-max** [yuewan2/MolMCL#5]
Your work is truly impressive! I do have a couple of questions:
In your code, it seems that during pre-training you did not use a validation set or a test set. Could you share why this choice was made, and how you ensured the model did not overfit under this setting?
Regarding the four different loss terms used during training, could you share the approximate numerical values of each loss? Are they similar, or do they differ significantly?

**Comment by code666-max** [comment]
after pretrain：
<img width="876" height="31" alt="Image" src="https://github.com/user-attachments/assets/3cba358d-d66f-4a19-be16-41b0d8ad8305" />
before pretrain：
<img width="891" height="42" alt="Image" src="https://github.com/user-attachments/assets/59c13ab0-1aaa-440a-9a5e-bdad9b8a54dd" />

This is the result I computed — it should be correct, right? I was a bit confused because, to me, loss1 was already as low as 0.4 before training, which seemed quite small.

**Comment by yuewan2** [comment]
Thank you for your interest! The short answer for not using an validation set for pretraining is that the finetune performance does not differ that much from our experiments. We hypothesize that since pretraining only offers a starting point of finetuning, the exact generalizability (whether it overfits) of learned knowledge may not matter that much. This is also one of the reason why most pretraining works (beyond this domain) often set dropout to 0. Also, it is tricky to evaluate validation performance of constrastive learning since the loss value depends on how samples are collected in batches. 

The loss you uploaded looks fine. loss_1 and loss_2 come from triplet loss with normalized distance so they should be small even before pretraining. I believe the main reason for loss_3 being large is the cross entropy loss of context prediction, which is also expected.

**Comment by code666-max** [comment]
Thank you for your detailed explanation — I’ve learned a lot from it. I still have two follow-up questions:

1. Regarding your point “The short answer for not using a validation set for pretraining is that the finetune performance does not differ that much from our experiments”, could you kindly share some related references or literature? I would like to read more about this topic.

2. For initializing the prompt weights using the roughness index (ROGI), does it produce three values similar to [0.45, 0.1, 0.45] as in:
preset_weight_1 = torch.FloatTensor([0.45, 0.1, 0.45]).to(graph_value.device)

Thank you again for your time and help.

**Comment by yuewan2** [comment]
Unfortunately I don't have a direct reference for the first point. It is mainly from past experience. For prompt weight initialization at finetuning, it does not necessarily produce similar preset values as pretraining. It really depends on finetune tasks and it can differ by a lot.

**Comment by code666-max** [comment]
Thank you very much for your answer, and sorry for the late reply.
